mpd/mpd.go

```diff
@@ -250,31 +254,70 @@ func (as *AdaptationSet) AddNewContentProtectionRoot(defaultKIDHex string) (*CEN
 	return cp, nil
 }
 
-// Adds a new content protection scheme for Widevine DRM.
-// This does not include the PSSH headers in the Manifest.
-func (as *AdaptationSet) AddNewContentProtectionSchemeWidevine(pssh *string) (*WidevineContentProtection, error) {
-	if pssh != nil {
-		if _, err := base64.StdEncoding.DecodeString(*pssh); err != nil {
-			return nil, err
-		}
+// AddNewContentProtectionSchemeWidevine adds a new content protection scheme for Widevine DRM to the adaptation set.
+// If wvHeader is empty or nil then the ContentProtection element will contain a <cenc:pssh> element which is a base64
+// representation of the PSSH box that would be found in the media init segment.
+// wvHeader - can be at most 1 byte slice
+// !!! Note: this function will accept any wvHeader value !!!
+func (as *AdaptationSet) AddNewContentProtectionSchemeWidevine(wvHeaders ...[]byte) (*WidevineContentProtection, error) {
+	if len(wvHeaders) > 1 {
+		panic("AddNewContentProtectionSchemeWidevine accepts at most 1 wvHeader slice")
```

Comment:
For backwards compatibility. It makes wvHeaders optional, if more than 1 argument is passed to the function it's more of a syntax error, rather than something that can/should be handled by the applicaiton.